# Import Library

Menginstal library dan modul yang akan digunakan

In [1]:
!pip install -q split-folders

Menghubungkan Google Drive dengan Google Colab

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# Import Data from Drive

Mengakses dataset dari Google Drive

In [3]:
dataset = '/content/drive/MyDrive/Dataset/Balance Breast Cancer Dataset'

Melakukan pembagian data dengan ratio 90% (train) dan 10% (test)

In [5]:
#impor package split-folders
import splitfolders

#Membagi data
splitfolders.ratio(dataset,
                   output='/content/drive/MyDrive/Dataset/Balance Breast Cancer Dataset/data',
                   seed=1337,
                   ratio=(.9, .1),
                   group_prefix=None, move=False)


Copying files: 0 files [00:00, ? files/s]
Copying files: 4 files [00:00, 35.75 files/s]
Copying files: 11 files [00:00, 50.33 files/s]
Copying files: 17 files [00:00, 50.70 files/s]
Copying files: 23 files [00:00, 52.87 files/s]
Copying files: 29 files [00:00, 49.30 files/s]
Copying files: 35 files [00:00, 51.33 files/s]
Copying files: 41 files [00:00, 50.82 files/s]
Copying files: 47 files [00:00, 52.47 files/s]
Copying files: 53 files [00:01, 51.96 files/s]
Copying files: 60 files [00:01, 55.00 files/s]
Copying files: 66 files [00:01, 52.97 files/s]
Copying files: 73 files [00:01, 55.88 files/s]
Copying files: 79 files [00:01, 54.36 files/s]
Copying files: 85 files [00:01, 51.68 files/s]
Copying files: 91 files [00:01, 46.10 files/s]
Copying files: 97 files [00:01, 48.92 files/s]
Copying files: 103 files [00:02, 50.06 files/s]
Copying files: 109 files [00:02, 50.97 files/s]
Copying files: 115 files [00:02, 51.94 files/s]
Copying files: 121 files [00:02, 53.04 files/s]
Copying files:

# Split Data

In [8]:
import os

base_dir = '/content/drive/MyDrive/Dataset/Balance Breast Cancer Dataset/data'

train_dir = os.path.join(base_dir, 'train')
os.chdir(base_dir)
os.rename("val","test")
test_dir = os.path.join(base_dir, 'test')

train_benign = '/content/drive/MyDrive/Dataset/Balance Breast Cancer Dataset/data/train/benign'
train_malignant = '/content/drive/MyDrive/Dataset/Balance Breast Cancer Dataset/data/train/malignant'
train_normal = '/content/drive/MyDrive/Dataset/Balance Breast Cancer Dataset/data/train/normal'

test_benign = '/content/drive/MyDrive/Dataset/Balance Breast Cancer Dataset/data/test/benign'
test_malignant = '/content/drive/MyDrive/Dataset/Balance Breast Cancer Dataset/data/test/malignant'
test_normal = '/content/drive/MyDrive/Dataset/Balance Breast Cancer Dataset/data/test/normal'

print('total citra train:', len(os.listdir(train_benign))+
                            len(os.listdir(train_malignant))+
                            len(os.listdir(train_normal))
                            )
print('total citra test:', len(os.listdir(test_benign))+
                           len(os.listdir(test_malignant))+
                           len(os.listdir(test_normal)))

total citra train: 431
total citra test: 49


Hasil pembagian data citra yang digunakan untuk train sebanyak 431 citra dan citra yang digunakan untuk test sebanyak 49 citra. Data ini digunakan untuk analisis data balance